hw/block/nvme.c

```diff
@@ -3049,6 +3049,7 @@ static void nvme_instance_init(Object *obj)
                         nvme_get_bootindex,
                         nvme_set_bootindex, NULL, NULL, NULL);
     object_property_set_int(obj, -1, "bootindex", NULL);
+    object_ref(obj);
```

Comment:
When is this released again?

Reply:
It is released as part of the pcie device removal. You can do this by issuing device_del $NAME from the qemu monitor. Note that you need to name the device on the config file in order to be able to do this. If the reference is not taken, then removing the device dereferences a NULL pointer.

This is necessary when testing metadata state (e.g., for recovery), which is saved on lnvm exit. This patch is not triggered when shutting down the machine, unfortunately.